optuna/storages/_base.py

```diff
@@ -162,14 +163,15 @@ def set_study_system_attr(self, study_id: int, key: str, value: Any) -> None:
         raise NotImplementedError
 
     @abc.abstractmethod
-    def set_study_direction(self, study_id: int, direction: StudyDirection) -> None:
+    def set_study_directions(self, study_id: int, directions: Sequence[StudyDirection]) -> None:
```

Comment:
I changed the name and the type of the argument without any deprecation. Since `optuna.storages` is not intended to be touched directly by users, I decided it wasn't a problem.